app/src/main/java/com/asfoundation/wallet/viewmodel/ConfirmationViewModel.kt

```diff
@@ -0,0 +1,99 @@
+package com.asfoundation.wallet.viewmodel
+
+import android.app.Activity
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.asfoundation.wallet.entity.GasSettings
+import com.asfoundation.wallet.entity.PendingTransaction
+import com.asfoundation.wallet.entity.TransactionBuilder
+import com.asfoundation.wallet.interact.FetchGasSettingsInteract
+import com.asfoundation.wallet.interact.SendTransactionInteract
+import com.asfoundation.wallet.logging.Logger
+import com.asfoundation.wallet.router.GasSettingsRouter
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+
+class ConfirmationViewModel internal constructor(
+    private val sendTransactionInteract: SendTransactionInteract,
+    private val gasSettingsRouter: GasSettingsRouter,
+    private val gasSettingsInteract: FetchGasSettingsInteract,
+    private val logger: Logger) :
+    BaseViewModel() {
+  private val transactionBuilder =
+      MutableLiveData<TransactionBuilder>()
+  private val transactionHash =
+      MutableLiveData<PendingTransaction>()
+  private var subscription: Disposable? = null
+
+  companion object {
+    private val TAG = ConfirmationViewModel::class.java.simpleName
+  }
+
+  fun init(transactionBuilder: TransactionBuilder) {
+    subscription = gasSettingsInteract.fetch(transactionBuilder.shouldSendToken())
+        .doOnSuccess { gasSettings: GasSettings? ->
+          transactionBuilder.gasSettings(gasSettings)
+          this.transactionBuilder.postValue(transactionBuilder)
+        }
+        .subscribe({}, { throwable: Throwable -> onError(throwable) })
+  }
+
+  override fun onCleared() {
+    if (subscription != null && !subscription!!.isDisposed) {
+      subscription!!.dispose()
+    }
```

Comment:
Use let. Plus I don't why kotlin doesn't say that we don't need to use !! because we are checking for null before

Reply:
Done